mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/LinkBankAccountViewModel.kt

```diff
@@ -0,0 +1,118 @@
+import org.mifos.mobilewallet.mifospay.R
+import org.mifos.mobilewallet.mifospay.common.Constants
+import org.mifos.mobilewallet.mifospay.domain.model.Bank
+import org.mifos.mobilewallet.mifospay.utils.FileUtils
+import java.util.Random
+import javax.inject.Inject
+
+
+@HiltViewModel
+class LinkBankAccountViewModel @Inject constructor() : ViewModel() {
+
+
+    private val _popularBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var popularBanks = _popularBanks.asStateFlow()
+
+    private val _allBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var allBanks = _allBanks.asStateFlow()
+
+    private val _filteredBanks = MutableStateFlow<List<Bank>>(emptyList())
+    var filteredBanks = _filteredBanks.asStateFlow()
+
+    private val _bankState = MutableStateFlow<BankUiState>(BankUiState.Loading)
+    val bankState = _bankState.asStateFlow()
+
+
+    fun getAllBanks(context: Context) {
+
+        _bankState.value = BankUiState.Loading
+
+        val popularBanks = ArrayList<Bank>()
+        popularBanks.add(Bank("RBL Bank", R.drawable.logo_rbl, 0))
+        popularBanks.add(Bank("SBI Bank", R.drawable.logo_sbi, 0))
+        popularBanks.add(Bank("PNB Bank", R.drawable.logo_pnb, 0))
+        popularBanks.add(Bank("HDFC Bank", R.drawable.logo_hdfc, 0))
+        popularBanks.add(Bank("ICICI Bank", R.drawable.logo_icici, 0))
+        popularBanks.add(Bank("AXIS Bank", R.drawable.logo_axis, 0))
+
+        _popularBanks.value = popularBanks
+
+        val jsonObject: JSONObject
+        jsonObject = FileUtils.readJson(context, "banks.json")!!
+        val banksList = ArrayList<Bank>()
+        for (i in 0 until jsonObject.getJSONArray("banks").length()) {
+            banksList.add(
+                Bank(
+                    (jsonObject.getJSONArray("banks")[i] as String), R.drawable.ic_bank, 1
+                )
+            )
+        }
+        _allBanks.value = banksList
+        _filteredBanks.value = banksList
+        _bankState.value = BankUiState.LoadedBankState(
+            popularBanks = _popularBanks.value,
+            allBanks = _filteredBanks.value
+        )
+
+    }
+
+    fun filterBanks(query: String) {
+        val filteredList = _allBanks.value.filter {
+            it.name.contains(query, ignoreCase = true)
+        }
+        if (query.isBlank()) {
+            _filteredBanks.value = _allBanks.value
+        } else {
+            _filteredBanks.value = filteredList
+        }
+        _bankState.value = BankUiState.LoadedBankState(
+            popularBanks = _popularBanks.value,
+            allBanks = _filteredBanks.value
+        )
+    }
+
+    fun fetchBankAccountDetails(bankName: String?) {
+        // TODO:: UPI API implement
+        addBankAccount(
+            BankAccountDetails(
+                bankName, "Ankur Sharma", "New Delhi",
+                mRandom.nextInt().toString() + " ", "Savings"
+            )
+
+
+        )
+    }
+
+    fun addBankAccount(bankAccountDetails: BankAccountDetails?) {
```

Comment:
expose this in LinkBankAccountActivity, we will consume later in compose navigating graph.